### PR TITLE
[Property grid]: update deps

### PR DESCRIPTION
**Pull request by MartynasStrazdas**
closes https://github.com/iTwin/viewer-components-react/issues/1059

### CODE REVIEW

File: change/@itwin-property-grid-react-8367431f-b9d7-46b2-86c8-533d9857b0ae.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "none",
+  "comment": "DevDependency update",
+  "packageName": "@itwin/property-grid-react",
+  "email": "43886789+MartynasStrazdas@users.noreply.github.com",
+  "dependentChangeType": "none"
```
Comment (MartynasStrazdas) [inline change/@itwin-property-grid-react-8367431f-b9d7-46b2-86c8-533d9857b0ae.json:6]: I'm not sure what type the change should be (none, patch or minor)? @saskliutas @grigasp

Comment (grigasp) [reply change/@itwin-property-grid-react-8367431f-b9d7-46b2-86c8-533d9857b0ae.json:6]: If there are no api or behavior changes, then it's a patch.

**Review by grigasp — COMMENTED** [review]
Any reason not to bump `@itwin/itwinui-react` to latest?

File: packages/itwin/property-grid/src/hooks/UseContextMenu.tsx [hunk 2]
```diff
@@ -92,9 +92,16 @@ export interface FavoritePropertiesContextMenuItemProps extends DefaultContextMe
...
+  const [hasFavorite, setHasFavorite] = useState(false);
+
+  useEffect(() => {
+    field &&
+      Presentation.favoriteProperties.hasAsync(field, imodel, currentScope).then((has) => {
+        return setHasFavorite(!has);
+      });
+  }, [currentScope, field, imodel]);
+
+  if (!hasFavorite || !field) return null;
```
Comment (grigasp) [inline packages/itwin/property-grid/src/hooks/UseContextMenu.tsx:104]:
```suggestion
  if (!hasFavorite || !field) {
    return null;
  }
```

File: packages/itwin/property-grid/src/hooks/UseContextMenu.tsx [hunk 3]
```diff
@@ -122,9 +129,16 @@ export function AddFavoritePropertyContextMenuItem({ field, imodel, scope, onSel
...
+  const [hasFavorite, setHasFavorite] = useState(false);
+
+  useEffect(() => {
+    field &&
+      Presentation.favoriteProperties.hasAsync(field, imodel, currentScope).then((has) => {
+        return setHasFavorite(has);
+      });
+  }, [currentScope, field, imodel]);
+
+  if (!hasFavorite || !field) return null;
```
Comment (grigasp) [inline packages/itwin/property-grid/src/hooks/UseContextMenu.tsx:141]:
```suggestion
  if (!hasFavorite || !field) {
    return null;
  }
```

**Review by grigasp — COMMENTED** [review]
LGTM, but I'd like @saskliutas to review as well :)

File: packages/itwin/property-grid/src/hooks/UseResizeObserver.ts [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+import { useCallback, useRef, useState } from "react";
+import { assert } from "@itwin/core-bentley";
+
+/**
+ * @internal
+ */
+export function useResizeObserver<T extends HTMLElement>() {
+  const observer = useRef<ResizeObserver>();
+  const [{ width, height }, setSize] = useState<{ width: number; height: number }>({ width: 0, height: 100 });
```
[the remark this answers is not in the document]
Comment (MartynasStrazdas) [reply packages/itwin/property-grid/src/hooks/UseResizeObserver.ts]: left over from testing, changed to 0.

File: packages/itwin/property-grid/src/hooks/UseContextMenu.tsx [hunk 5]
```diff
@@ -202,7 +220,12 @@ export function useContextMenu({ dataProvider, imodel, contextMenuItems }: UseCo
     }
 
     const field = await dataProvider.getFieldByPropertyDescription(args.propertyRecord.property);
-    const items = contextMenuItems.map((item, index) => <Fragment key={index}>{item({ imodel, dataProvider, record: args.propertyRecord, field })}</Fragment>);
+    const items = (close: () => void) =>
+      contextMenuItems.map((item, index) => (
+        <Flex key={index} onClick={close}>
```
[the remark this answers is not in the document]
Comment (saskliutas) [reply packages/itwin/property-grid/src/hooks/UseContextMenu.tsx:225]: I don't think we should support arbitrary components as menu items. We provide `PropertyGridContextMenuItem` component as a base for custom menu items. It ensures consistent styling and behavior so if consumer decides to use arbitrary components we cannot guarantee that everything will work.

[the remark this answers is not in the document]
Comment (MartynasStrazdas) [reply packages/itwin/property-grid/src/hooks/UseContextMenu.tsx]: Changed

File: packages/itwin/property-grid/src/hooks/UseContextMenu.tsx [hunk 6]
```diff
@@ -220,12 +227,7 @@ export function useContextMenu({ dataProvider, imodel, contextMenuItems }: UseCo
     }
 
     const field = await dataProvider.getFieldByPropertyDescription(args.propertyRecord.property);
-    const items = (close: () => void) =>
-      contextMenuItems.map((item, index) => (
-        <Flex key={index} onClick={close}>
-          {item({ imodel, dataProvider, record: args.propertyRecord, field })}
-        </Flex>
-      ));
+    const items = contextMenuItems.map((item, index) => <Flex key={index}>{item({ imodel, dataProvider, record: args.propertyRecord, field })}</Flex>);
```
[the remark this answers is not in the document]
Comment (MartynasStrazdas) [reply packages/itwin/property-grid/src/hooks/UseContextMenu.tsx]: fixed

File: packages/itwin/property-grid/src/hooks/UseContextMenu.tsx [hunk 7]
```diff
@@ -259,3 +263,23 @@ export function useContextMenu({ dataProvider, imodel, contextMenuItems }: UseCo
...
+  const [value] = useState(() => ({
+    close,
+  }));
+
+  return <contextMenuContext.Provider value={value}>{children}</contextMenuContext.Provider>;
+}
+
+function useContextMenuContext() {
+  return useContext(contextMenuContext);
+}
```
[the remark this answers is not in the document]
Comment (MartynasStrazdas) [reply packages/itwin/property-grid/src/hooks/UseContextMenu.tsx]: fixed

**Review by saskliutas — APPROVED** [review]

**Pull request merged**